Source/DafnyCore/Generic/Node.cs

```diff
@@ -30,6 +30,16 @@ public class SyntaxConstructorAttribute : Attribute { }
 [AttributeUsage(AttributeTargets.Parameter | AttributeTargets.Field)]
 public class BackEdge : Attribute { }
 
+/// <summary>
+/// Indicates that the target field or property need not be serialized.
+/// This is useful when a base type defines a field or property,
+/// and all instances of a subtype use the same particular value for that inherited field or property.
+/// </summary>
+[AttributeUsage(AttributeTargets.Class, AllowMultiple = true)]
+public class NonSerializedField(string name) : Attribute {
```

Comment:
I don't like the name, because this property is more general than serialization. You can also use this property for example to customize how you generate a printer for a type. I'd prefer a name like `NonInheritedField`

Reply:
It's certainly possible to use this attribute to customize printing, but unless we plan to do so, I'd rather not expand its scope preemptively. I'm open to other suggestions, though I don't like `NotInheritedField` because it's not true (at least, in the intended use case). What do you think about something like `ConstantInheritedField` or `FixedInheritedField`?

Reply:
What about `ComputedInheritedField` ? To indicate that its value is determined based on the other fields.

Reply:
Based on offline discussion, I've renamed `NonSerializedField` to `RedundantField`.